Initial commit of syntax highlighting VS Code extension

rosemwahlin opened the pull request:
Initial commit of syntax highlighting.

README points to a file in the rticommunity repo that won't be there until we merge.  (If the URL is wrong, it might need to be update)

rosemwahlin committed:
Initial commit of syntax highlighting VS Code extension

rosemwahlin committed:
README fix

fgaranda commented on package.json:
```diff
@@ -0,0 +1,25 @@
+    "name": "omg-idl-lang",
+    "displayName": "Interface Definition Lang",
+    "description": "Syntax highlighting for OMG IDL language",
+    "version": "0.0.1",
```
Should this be version 1.0?

fgaranda commented on syntaxes/omg.idl.tmLanguage.json:
```diff
@@ -0,0 +1,567 @@
+{
```
Could you use 4-spaces instead of tabs? The rest of documents are using that convention

fgaranda commented on syntaxes/omg.idl.tmLanguage.json:
```diff
@@ -0,0 +1,567 @@
+		"signed_short_int": {
+			"patterns": [{
+				"name": "storage.type.primitive.omg.idl",
+				"match": "\\b(short)\\b"
```
This pattern should also match the new fixed-width integer notation defined in the Extended Types building block of IDL 4.2. That is, it should match both `short` and `int16`.

The same applies to the rest of signed and unsigned integer types:
- `unsigned short` - `uint16`
- `long` - `int32`
- `unsigned long` - `uint32`
- `long long` - `int64`
- `unsigned long long`  - `uint64`

fgaranda commented on syntaxes/omg.idl.tmLanguage.json:
```diff
@@ -0,0 +1,567 @@
+				{"include": "#signed_short_int"},
+				{"include": "#signed_long_int"},
+				{"include": "#signed_longlong_int"}
+			]
```
IDL 4.2 has introduced support for 8-bit integer values. They are definde in BNF as follows:
```
(208) <signed_tiny_int> ::= “int8”
(209) <unsigned_tiny_int> ::= “uint8”
```
Following your notation, you could add an include for `signed_tiny_int` under `signed_int` and `unsigned_tiny_int` under `unsigned_int`.

fgaranda commented on syntaxes/omg.idl.tmLanguage.json:
```diff
@@ -0,0 +1,567 @@
+		
+	},
+	"scopeName": "source.omg.idl"
+}
```
Add newline

rosemwahlin committed:
Whitespace fixes

rosemwahlin committed:
Added missing storage types, updated version to 1.0.0

rosemwahlin committed:
Fixing issues where:
    - modules expecting to start at the beginning of a line
    - structs not recognized when using inheritance
    - case statements expecting leading whitespace

rosemwahlin commented:
Issues should be fixed, including the ones visible in the test file you recommended.

rosemwahlin committed:
Fixing issue where numerics in case labels were highlighted incorrectly when part of an identifier

rosemwahlin committed:
Fixing issues that prevented packaging extension
